skaffold init --skip-build error

<!--
  Issues without logs and details are more complicated to fix.
  Please help us by filling the template below!
-->

### Expected behavior
generate skaffold.yaml file

### Actual behavior
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x513b3a5]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.(*builderAnalyzer).enterDir(0x0, 0x5584c75, 0x1)
        <autogenerated>:1 +0x5
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.(*analysis).analyze(0xc0005e7730, 0x5584c75, 0x1, 0x80, 0xc000291de0)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/analyze.go:63 +0xfb
github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer.DoInit(0x589dc40, 0xc00050ad40, 0x5857de0, 0xc0000c6008, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/pkg/skaffold/initializer/init.go:86 +0x14e
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doInit(0x589dc40, 0xc00050ad40, 0x5857de0, 0xc0000c6008, 0xc000491a00, 0xf)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/init.go:70 +0x1db
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.cancelWithCtrlC.func1(0x5857de0, 0xc0000c6008, 0x0, 0x0)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/signals.go:33 +0xe1
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0xc00029b180, 0xc000491960, 0x0, 0x1, 0x0, 0x0)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd/commands.go:96 +0x6a
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).execute(0xc00029b180, 0xc000491940, 0x1, 0x1, 0xc00029b180, 0xc000491940)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:826 +0x460
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc00041e000, 0xc0000c6008, 0x5857de0, 0xc0000c6010)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra.(*Command).Execute(...)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/vendor/github.com/spf13/cobra/command.go:864
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x5857de0, 0xc0000c6008, 0x5857de0, 0xc0000c6010, 0xc0005e7f50, 0x400759f)
        /ext-go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/skaffold.go:27 +0x57
main.main()
        /ext-go/src/github.com/GoogleContainerTools/skaffold/cmd/skaffold/skaffold.go:30 +0x4e

### Information

- Skaffold version: 1.3.1
- Operating system: macOS Catalina 10.15.3
- Contents of skaffold.yaml:

```yaml
<paste your skaffold.yaml here>
```

### Steps to reproduce the behavior

1. a clonable repository with the sample skaffold project
2. `skaffold init --skip-build`

--- a/pkg/skaffold/initializer/analyze.go
+++ b/pkg/skaffold/initializer/analyze.go
@@ -35,18 +35,14 @@ type analysis struct {
 
 // newAnalysis sets up the analysis of the directory based on the initializer configuration
 func newAnalysis(c Config) *analysis {
-	var builders *builderAnalyzer
-	if !c.SkipBuild {
-		builders = &builderAnalyzer{
+	return &analysis{
+		kubectlAnalyzer: &kubectlAnalyzer{},
+		builderAnalyzer: &builderAnalyzer{
 			findBuilders:         !c.SkipBuild,
 			enableJibInit:        c.EnableJibInit,
 			enableBuildpacksInit: c.EnableBuildpacksInit,
 			buildpacksBuilder:    c.BuildpacksBuilder,
-		}
-	}
-	return &analysis{
-		kubectlAnalyzer: &kubectlAnalyzer{},
-		builderAnalyzer: builders,
+		},
 		skaffoldAnalyzer: &skaffoldConfigAnalyzer{
 			force:        c.Force,
 			analyzeMode:  c.Analyze,

--- a/pkg/skaffold/initializer/analyze_test.go
+++ b/pkg/skaffold/initializer/analyze_test.go
@@ -73,6 +73,34 @@ func TestAnalyze(t *testing.T) {
 			shouldErr: false,
 		},
 		{
+			description: "--skip-build should return no builders in analysis",
+			filesWithContents: map[string]string{
+				"config/test.yaml":       validK8sManifest,
+				"config/invalid.yaml":    emptyFile,
+				"k8pod.yml":              validK8sManifest,
+				"README":                 emptyFile,
+				"deploy/Dockerfile":      emptyFile,
+				"deploy/Dockerfile.dev":  emptyFile,
+				"deploy/dev.Dockerfile":  emptyFile,
+				"deploy/test.dockerfile": emptyFile,
+				"gradle/build.gradle":    emptyFile,
+				"maven/pom.xml":          emptyFile,
+				"Dockerfile":             emptyFile,
+			},
+			config: Config{
+				Force:                false,
+				EnableBuildpacksInit: false,
+				EnableJibInit:        false,
+				SkipBuild:            true,
+			},
+			expectedConfigs: []string{
+				"k8pod.yml",
+				"config/test.yaml",
+			},
+			expectedPaths: []string{},
+			shouldErr:     false,
+		},
+		{
 			description: "should return correct k8 configs and build files",
 			filesWithContents: map[string]string{
 				"config/test.yaml":    validK8sManifest,

